server/routes/user.ts

```diff
@@ -184,19 +188,26 @@ router.post('/import-from-plex', async (req, res, next) => {
     for (const rawUser of plexUsersResponse.MediaContainer.User) {
       const account = rawUser.$;
       const user = await userRepository.findOne({
-        where: { plexId: account.id },
+        where: { email: account.email },
       });
       if (user) {
         // Update the users avatar with their plex thumbnail (incase it changed)
         user.avatar = account.thumb;
         user.email = account.email;
-        user.username = account.username;
+        user.plexUsername = account.username ?? account.email;
+
+        // in-case the user was previously a local account
+        if (user.userType === UserType.LOCAL) {
+          user.userType = UserType.PLEX;
+          user.username = '';
```

Comment:
Do we need to reset the username? We should probably just leave it as whatever it was set to.

Reply:
If you don't want to reset this in all cases, we should at least reset it whenever the username has the same value as email. What do you think? Otherwise local users without usernames set, will not get an eventual username set in plex. Updates to plex username won't be used either.

Reply:
I've pushed these changes and fixed the other concern you had + rebased. If you think it's fine you should be able to merge. If you want me to fix anything I'll probably be available a few hours more.

Reply:
I think it should be expected that if a user sets a username for themselves, it won't be overwritten. We can eventually make it clear through the UI that you can clear your username to just use plex's username.

Reply:
```          // reset data (as it now resides in plexUsername)
          // if an explicit username has not been set
          if (user.username === user.email) {
            user.username = '';
          }```

It only clears the username field if they haven't had their username edited since local user creation.

Reply:
I agree that once set, the username shouldn't be overwritten.  But can we add a reset button of some sort, in case a user wants their username to be synced with Plex again?  Maybe it could be like the button to regenerate the API key in General Settings.

Reply:
It isn't overwritten and if we want to have this logic more complex I suggest we add it to the backlog.

Reply:
@ankarhem Couldn't the button just set the `username` to an empty string, if that signifies that the value should be read from `plexUsername`?